### ISSUE TITLE
Keycloak version >= 24 support?

### ISSUE THREAD

**Issue by sventorben** [keycloak/terraform-provider-keycloak#944]
I'm reaching out to see if there are any plans to update the Keycloak Terraform provider for Keycloak version 24 support.

Many of us are eager to take advantage of the new features in version >= 24 and integrate them into our Terraform workflows.
Could you share if this is on the roadmap, and if so, any expected timelines? Also, if there's a way for the community to stay updated on progress or plans, that would be great to know.

Appreciate all the work you do on this provider. It's been incredibly helpful for our projects.

**Comment by bradhead** [comment]
Now that the user profile is pretty much mandatory, the 'keycloak_realm_user_profile' resource type is essential to setup any managed attributes beyond the default.   Right now with 24.0.3, when I try to use this, it complains that you can't remove username or email... in other words, it seems to try to do a POST perhaps to the Admin REST API instead of a PUT to update the existing profile.   Profile is an oddity as there is just a single resource, pre-existing now, so an import should/could be assumed.

**Comment by ofairfoul** [comment]
also it's missing the new multivalued flag for attributes in KC24.

**Comment by grolingm-VU** [comment]
Hello everyone,

if not done already please participate in the latest survey from the Keycloak team [https://www.keycloak.org/2024/06/realm-config-manamagemtn-tools-survey](https://www.keycloak.org/2024/06/realm-config-manamagemtn-tools-survey).
This could give this provider the needed gravity.

**Comment by marijapopovikj-casumo** [comment]
@sventorben have you used this provider with keycloak version > =23.0.0 ? We are pushing towards updating keycloak to newer versions (we are at the moment at 22.0.0 testing it at the moment with 23.0.0) and not sure if this provider works with higher versions than this, or if it will support the latest keycloak versions.

**Comment by sventorben** [comment]
> @sventorben have you used this provider with keycloak version > =23.0.0 ? We are pushing towards updating keycloak to newer versions (we are at the moment at 22.0.0 testing it at the moment with 23.0.0) and not sure if this provider works with higher versions than this, or if it will support the latest keycloak versions.

Yes we do use it with KC 24.0.5. It is not supporting the latest features though.

**Comment by marijapopovikj-casumo** [comment]
@sventorben  have you tried to run the tests of this repo (as desrcibed in the README) with different keycloak versions? When I've tried to execute them I got couple of failing even with the default versions specified   
 21.0.1, 20.0.5 , 19.0.2.  
 Like this one
 ```
resource_keycloak_custom_user_federation_test.go 
   Error: custom user federation provider with id custom is not installed on the server
        
          with keycloak_custom_user_federation.custom,
          on terraform_plugin_test.tf line 6, in resource "keycloak_custom_user_federation" "custom":
           6: resource "keycloak_custom_user_federation" "custom" {
}
```

**Comment by sventorben** [comment]
No, we didn't execute any of the tests. I am not really familiar with Go programming and use the provider more like a pre-packages artifact and not trying to develop it or extend it. There was not really a need for me to execute the tests.

**Comment by mkyc** [comment]
There is promising PR https://github.com/mrparkers/terraform-provider-keycloak/pull/987 it would be awesome to have it, as we're stopped as well.
